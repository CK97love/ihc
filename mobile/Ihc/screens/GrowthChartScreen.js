import data from '../services/DataService';
import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Button,
  Text,
  ScrollView,
  View
} from 'react-native';
import ScatterPlot from '../components/ScatterPlot'
const boysWeightData = require('../growthchartdata/boys_weights.json');
const girlsWeightData = require('../growthchartdata/girls_weights.json');
const boysHeightData = require('../growthchartdata/boys_heights.json');
const girlsHeightData = require('../growthchartdata/girls_heights.json');

const PLOT_HEIGHT = 400;
const PLOT_WIDTH = 400;
const PLOTS_HEIGHT = PLOT_HEIGHT * 2 + 150;

export default class GrowthChartScreen extends Component<{}> {
  /*
   * Gets a patientInfo prop
   */
  constructor(props) {
    super(props);
  }

  extractData(data) {
    const arr = [];
    arr.push({ color: 'red', unit: '%', values: data["P3"]})
    arr.push({ color: 'orange', unit: '%', values: data["P5"]})
    arr.push({ color: 'purple', unit: '%', values: data["P10"]})
    arr.push({ color: 'green', unit: '%', values: data["P25"]})
    arr.push({ color: 'blue', unit: '%', values: data["P50"]})
    arr.push({ color: 'purple', unit: '%', values: data["P75"]})
    arr.push({ color: 'orange', unit: '%', values: data["P90"]})
    arr.push({ color: 'red', unit: '%', values: data["P95"]})
    return arr;
  }

  render() {
    // TODO: show boy or girl dependent on person
    const weightData = this.extractData(girlsWeightData);
    const heightData = this.extractData(girlsHeightData);
    // TODO: add data for this person's height/weight

    // Mark every 2 yrs
    const arrYears = Array.from({length: 10}, (v,i) => i * 24 ); 
    // Mark every 25 cm
    const arrCm = Array.from({length: 10}, (v,i) => i * 25 ); 
    // Mark every 25 kg
    const arrKg = Array.from({length: 5}, (v,i) => i * 25 ); 

    return (
      // TODO: Label the grid lines
      <ScrollView contentContainerStyle={styles.scrollContainer}>
        <Text style={styles.title}>Growth Chart</Text>

        <View style={styles.plotsContainer}>
          <View style={styles.plotContainer}>
            <ScatterPlot
              data={weightData}
              chartHeight={PLOT_HEIGHT}
              chartWidth={PLOT_WIDTH}
              minX={0}
              maxX={240}
              minY={0}
              maxY={125}
              horizontalLinesAt={arrKg}
              verticalLinesAt={arrYears}
              title="Weight Growth Chart"
              unitX='Age'
              unitY='Weight' />
          </View>

          <View style={styles.plotContainer}>
            <ScatterPlot
              chartHeight={PLOT_HEIGHT}
              chartWidth={PLOT_WIDTH}
              data={heightData}
              minX={0}
              maxX={240}
              minY={0}
              maxY={200}
              horizontalLinesAt={arrCm}
              verticalLinesAt={arrYears}
              title="Height Growth Chart"
              unitX='Age'
              unitY='Height' />
          </View>

        </View>
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  plotsContainer: {
    flex: 1,
    top: 50,
    margin: 8,
  },
  plotContainer: {
    height: PLOT_HEIGHT + 20,
    margin: 4,
    padding: 8,
    backgroundColor: '#ededed'
  },
  scrollContainer: {
    minHeight: PLOTS_HEIGHT,
    flex: 0,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
    position: 'absolute',
    top: 4
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
