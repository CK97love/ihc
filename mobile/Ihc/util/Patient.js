// Realm returns arrays as objects, so convert them before sending to server
export function convertObjectsToArrays(patient) {
  const copy = Object.assign({}, patient);
  copy.medications = Array.from(patient.medications);
  copy.soaps = Array.from(patient.soaps);
  copy.triages = Array.from(patient.triages);
  copy.statuses = Array.from(patient.statuses);
  return copy;
}
