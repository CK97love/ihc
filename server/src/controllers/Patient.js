import PatientModel from '../models/Patient';
import SoapModel from '../models/Soap';
import StatusModel from '../models/Status';
import TriageModel from '../models/Triage';
import DrugUpdateModel from '../models/DrugUpdate';

const PatientController = {
  GetPatient: function(req, res){
    PatientModel.findOne({key: req.params.key}, function(err, patient) {
      if(!patient) {
        err = new Error("Patient with key " + req.params.key + " doesn't exist");
      }
      if(err) {
        res.json({status: false, error: err.message});
        return;
      }
      res.json({status: true, patient: patient});
    });
  },
  GetPatients: function(req, res){
  },
  CreatePatient: function(req, res){
  },
  UpdatePatient: function(req, res){
  },
  GetUpdates: function(req, res){
  },
  GetSoap: function(req, res){
  },
  GetStatus: function(req, res){
  },
  GetTriage: function(req, res){
  },
  GetDrugUpdates: function(req, res){
  },
  UpdateSoap: function(req, res){
  },
  UpdateStatus: function(req, res){
  },
  UpdateTriage: function(req, res){
  },
  UpdateDrugUpdates: function(req, res){
  }
};

module.exports = PatientController;
