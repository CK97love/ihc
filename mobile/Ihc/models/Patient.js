import {stringDate} from '../util/Date';
export default class Patient {
  // Insert any class methods here
  get drugUpdates() {
    const drugToUpdates = {};
    for (var update of this.medications) {
      if (update.name in drugToUpdates) {
        drugToUpdates[update.name].push(update);
      } else{
        drugToUpdates[update.name] = [update];
      }
    }
    return drugToUpdates;
  }

  static fullName(patient) {
    return `${patient.firstName} ${patient.fatherName} ${patient.motherName}`;
  }

  // To be used as primary key
  static makeKey(patient) {
    const str = `${patient.firstName}&${patient.fatherName}&${patient.motherName}` +
      `&${patient.birthday}`;
    return str;
  }

  static extractFromForm(form) {
    const patient = Object.assign({}, form);
    patient.birthday = stringDate(form.birthday);
    patient.key = Patient.makeKey(patient);
    if(form.newPatient) {
      // 1 is male, 2 is female
      patient.gender = form.gender === 'Male' ? 1 : 2;
      patient.lastUpdated = new Date().getTime();
    }
    return patient;
  }
}

Patient.schema = {
  name: 'Patient',
  primaryKey: 'key',
  properties: {
    key: 'string',
    firstName: 'string',
    fatherName: 'string', // last name
    motherName: 'string', // last name
    birthday: 'string',
    gender: 'int', // 1 = boy, 2 = girl, 0 = undefined
    phone: 'string?',
    motherHeight: 'double?',
    fatherHeight: 'double?',
    medications: 'DrugUpdate[]',
    soaps: 'Soap[]',
    triages: 'Triage[]',
    statuses: 'Status[]',
    growthchart: 'GrowthChartUpdate[]',
    lastUpdated: 'int' // timestamp
  }
};
