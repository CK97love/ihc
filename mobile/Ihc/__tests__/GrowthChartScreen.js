import 'react-native';
import React from 'react';
import GrowthChartScreen from '../screens/GrowthChartScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import data from '../services/DataService';
import sinon from 'sinon';
import Patient from '../models/Patient';

it('renders correctly', () => {
  sinon.stub(data, 'getPatient').returns(Promise.resolve(Patient.getInstance()));
  const json = renderer.create(
    <GrowthChartScreen />
  ).toJSON();
  expect(json).toMatchSnapshot();
});
