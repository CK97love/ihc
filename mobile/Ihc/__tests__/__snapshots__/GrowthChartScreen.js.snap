// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 0,
      "justifyContent": "center",
      "minHeight": 950,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "margin": 10,
          "position": "absolute",
          "textAlign": "center",
          "top": 4,
        }
      }
    >
      Growth Chart
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "red",
          "margin": 10,
          "textAlign": "center",
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "margin": 8,
          "top": 50,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ededed",
            "height": 420,
            "margin": 4,
            "padding": 8,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "height": 400,
              "width": 400,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "margin": 4,
              }
            }
          >
            Weight Growth Chart
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "bottom": 100,
                "left": -8,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
              }
            }
          >
            Weight
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "bottom": 0,
                "left": 85,
                "position": "absolute",
              }
            }
          >
            Age
          </Text>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "red",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P95
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "orange",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P90
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "purple",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P75
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "blue",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P50
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "green",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P25
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "purple",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P10
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "orange",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P5
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "red",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P3
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              Patient
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ededed",
            "height": 420,
            "margin": 4,
            "padding": 8,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "height": 400,
              "width": 400,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "margin": 4,
              }
            }
          >
            Height Growth Chart
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "bottom": 100,
                "left": -8,
                "position": "absolute",
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
              }
            }
          >
            Height
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "bottom": 0,
                "left": 85,
                "position": "absolute",
              }
            }
          >
            Age
          </Text>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "red",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P95
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "orange",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P90
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "purple",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P75
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "blue",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P50
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "green",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P25
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "purple",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P10
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "orange",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P5
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "red",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              P3
            </Text>
          </View>
          <View
            style={
              Object {
                "alignContent": "center",
                "flexDirection": "row",
                "height": 20,
                "marginLeft": 4,
                "paddingTop": 4,
                "width": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "black",
                  "borderRadius": 5,
                  "bottom": undefined,
                  "height": 5,
                  "left": undefined,
                  "margin": 0,
                  "opacity": 0.3,
                  "position": "relative",
                  "width": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "bottom": 8,
                  "marginLeft": 4,
                  "maxHeight": 20,
                }
              }
            >
              Patient
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
